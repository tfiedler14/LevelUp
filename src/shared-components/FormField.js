import { Text, TextInput, View } from 'react-native';
import * as React from 'react';
import EStyleSheet from 'react-native-extended-stylesheet';

export const WrappedTextInput = props => {
  const { title, input, textContentType, autoCompleteType, password, multiline, meta, ...inputProps } = props;

  const error = meta.error;
  const touched = meta.touched;
  return (
    <View>
      <Text style={styles.title}>{title}</Text>
        <TextInput
          {...inputProps}
          onChangeText={input.onChange}
          onBlur={input.onBlur}
          onFocus={input.onFocus}
          style={styles.textEntry}
          textContentType={textContentType || 'none'}
          autoCompleteType={autoCompleteType || 'off'}
          secureTextEntry={password}
          multiline={multiline}
          maxLength={2500}
          value={input.value}
        />
      {touched && error && <Text style={styles.errorText}>{error}</Text>}
    </View>
  );
};



const styles = EStyleSheet.create({
  textEntry: {
    borderBottomColor: '#ccc',
    borderBottomWidth: '.1rem',
    backgroundColor: '#666',
    color: 'white',
    padding: '.5rem',
    marginRight: '1rem',
    marginLeft: '1rem',
    marginBottom: '.5rem'
  },
  title: {
    color: 'white',
    marginBottom: '.25rem'
  },
  errorText: {
    color: '#880000',
    fontSize: '.9rem',
    marginLeft: '1rem',
    marginTop: '-.25rem'
  }
});
