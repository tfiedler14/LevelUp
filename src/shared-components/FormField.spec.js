import React from 'react';
import { shallow } from 'enzyme';
import { WrappedTextInput } from './FormField';

describe('FormField', () => {
  it('should render FormField correctly', () => {
    const props = {
      title: 'test',
      input: {},
      textContentType: 'any',
      autoCompleteType: 'any',
      password: false,
      multiline: false,
      meta: {}
    };

    const component = shallow(<WrappedTextInput {...props} />);

    expect(component).toMatchSnapshot();
  });
});
