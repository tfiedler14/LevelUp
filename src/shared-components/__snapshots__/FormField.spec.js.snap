// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField should render FormField correctly 1`] = `
<View>
  <Text>
    test
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="any"
    maxLength={2500}
    multiline={false}
    rejectResponderTermination={true}
    secureTextEntry={false}
    textContentType="any"
    underlineColorAndroid="transparent"
  />
</View>
`;
